spawner.sh

```diff
@@ -153,22 +186,34 @@ done
 echo "Pushing metadata..."
 for slave in ${SLAVES[@]}; do
 	# The constructors should spin on es_spinlock_1 to avoid race conditions
-	gcloud compute instances add-metadata $slave \
-	--metadata es_slave_ips="${SLAVE_IPS[*]}",es_num_masters="$NUM_MASTERS",es_debug="$DEBUG",es_cluster_name="$CLUSTER_NAME",es_controller_ip="${PRIMARY_IP}",es_version="${ES_VERSION}",es_plugin_dir_version="${PLUGIN_DIR_VERSION}",es_plugin_version="${PLUGIN_VERSION}",es_logstash_version="${LOGSTASH_VERSION}",es_node_config="$ES_NODE_CONFIG",es_spinlock_1=released
+    # NB there must be NO WHITESPACE in the metadata string!
+	gcloud compute instances add-metadata $slave --metadata \
+es_slave_ips="${SLAVE_IPS[*]}",\
+es_num_masters="$NUM_MASTERS",\
+es_debug="$DEBUG",\
+es_cluster_name="$CLUSTER_NAME",\
+es_controller_ip="${PRIMARY_IP}",\
+es_version="${ES_VERSION}",\
+es_plugin_version="${PLUGIN_VERSION}",\
+es_logstash_version="${LOGSTASH_VERSION}",\
+es_node_config="${ES_NODE_CONFIG}",\
+es_download_url="${ES_DOWNLOAD_URL}",\
+custom_es_java_opts="${CUSTOM_ES_JAVA_OPTS}",\
+es_spinlock_1=released
 done
 
 echo "Waiting for OS to come up on each slave..."
-for slave in ${SLAVES[@]}; do
+for slave in ${SLAVE_IPS[@]}; do
 	while ! nc -w 5 $slave 22 </dev/null >/dev/null; do
 		sleep 5
 	done
-	echo "$slave running"
+	echo "ssh running on $slave"
 done
 # Repopulate known_hosts
-ssh-keyscan $SLAVES >> $HOME/.ssh/known_hosts
+#ssh-keyscan ${SLAVE_IPS[@]} >> $HOME/.ssh/known_hosts
```

Comment:
should this removed ?

Reply:
I'd really like to put this functionality back in iff I can find a way to do it without taking out the network. :-( Is it OK if I leave it there for now?

Reply:
sure!